### PR TITLE
Implement DPRKN5 method

### PR DESCRIPTION
**Pull request by ErikQQY**
Corresponding paper: [Bettis, D. G. "A Runge-Kutta Nyström algorithm." Celestial mechanics 8.2 (1973): 229-233.](https://link.springer.com/article/10.1007/BF01231421)

Convergence test:

err

The adaptive methods regression test fails, I think the implementation is correct since the algorithm convergence is fine, need some help on the adaptive methods regression tests🤔.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
What's the issue with the adaptive regression methods test? I don't see the test in there?

File: test/algconvergence/partitioned_methods_tests.jl [hunk 1]
```diff
@@ -158,6 +158,8 @@ sim = test_convergence(dts, prob_big, ERKN5(), dense_errors = true)
 @test sim.𝒪est[:L2]≈4 rtol=1e-1
 
 # Adaptive methods regression test
+sol = solve(prob, DPRKN5())
+@test length(sol.u) < 160
```
Comment (ChrisRackauckas) [inline test/algconvergence/partitioned_methods_tests.jl:162]: That seems a bit high. Are you sure the adaptivity parameters are correct?

Comment (ErikQQY) [reply test/algconvergence/partitioned_methods_tests.jl]: One of the adaptive parameters is wrong, just fixed!

File: src/tableaus/rkn_tableaus.jl [hunk 2]
```diff
@@ -313,6 +313,146 @@ function ERKN5ConstantCache(T::Type{<:CompiledFloats}, T2::Type{<:CompiledFloats
...
+end
+
+function DPRKN5ConstantCache(T::Type{<:CompiledFloats}, T2::Type{<:CompiledFloats})
+      c1 = convert(T2, 0.125)
+      c2 = convert(T2, 0.25)
+      c3 = convert(T2, 0.5)
+      c4 = convert(T2, 0.75)
+      c5 = convert(T2, 1.0)
+      a21 = convert(T, 0.0078125)
+      a31 = convert(T, 0.010416666666666666)
```
Comment (oscardssmith) [inline src/tableaus/rkn_tableaus.jl:421]: use rationals for these?

File: src/tableaus/rkn_tableaus.jl [hunk 3]
```diff
@@ -412,41 +412,45 @@ function DPRKN5ConstantCache(T::Type, T2::Type)
...
-    convert(T, 0.08888888888888889),
-    convert(T, 0.07777777777777778),
-    convert(T, 0.35555555555555557),
-    convert(T, 0.13333333333333333),
-    convert(T, 0.35555555555555557),
-    convert(T, 0.07777777777777778),
-    convert(T, -0.08888888888888888),
-    convert(T, 0.26666666666666666),
-    convert(T, -0.26666666666666666),
-    convert(T, 0.26666666666666666),
```
Comment (ChrisRackauckas) [inline src/tableaus/rkn_tableaus.jl]: hehe it happens

**Review by ChrisRackauckas — APPROVED** [review]

File: src/tableaus/rkn_tableaus.jl (hunk 2 above)
Comment (oscardssmith) [reply src/tableaus/rkn_tableaus.jl]: The reason is for better accuracy. If `T` is a `BigFloat` than `convert(T, 1//96)` will have way more correct digits.

File: src/tableaus/rkn_tableaus.jl [hunk 4]
```diff
@@ -574,14 +574,14 @@ function DPRKN5ConstantCache(T::Type{<:CompiledFloats}, T2::Type{<:CompiledFloat
     c3 = convert(T2, 0.5)
     c4 = convert(T2, 0.75)
     c5 = convert(T2, 1.0)
-    a21 = convert(T, 0.0078125)
-    a31 = convert(T, 0.010416666666666666)
+    a21 = convert(T, 1 // 128)
+    a31 = convert(T, 1 // 96)
     a32 = convert(T, 0.020833333333333332)
     a41 = convert(T, 0.041666666666666664)
     a43 = convert(T, 0.08333333333333333)
```
Comment (oscardssmith) [inline src/tableaus/rkn_tableaus.jl:581]: there are more rationals.

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
Thanks!

**Comment by ChrisRackauckas** [comment]
Can you PR to add to the docs?

**Comment by ErikQQY** [comment]
Sure, I will send a PR to add both ERKN7 and DPRKN5 to the docs

**Pull request merged**